client.go

```diff
@@ -0,0 +1,48 @@
+package id
+
+import (
+	"errors"
+	"github.com/emersion/go-imap"
+	"github.com/emersion/go-imap/client"
+)
+
+// Client is an ID client.
+type Client struct {
+	c *client.Client
+}
+
+// NewClient creates a new client.
+func NewClient(c *client.Client) *Client {
+	return &Client{c: c}
+}
+
+// SupportID checks if the server supports the ID extension.
+func (c *Client) SupportID() (bool, error) {
+	return c.c.Support(Capability)
+}
+
+func (c *Client) id(clientID ID) (serverID ID, err error) {
+	if state := c.c.State(); imap.ConnectedState&state != state {
+		return nil, errors.New("Not connected")
```

Comment:
You can use `client.ErrNotLoggedIn` here

Reply:
IMHO the message of `client.ErrNotLoggedIn` does not fit here, since the ID command is allowed in any stage, even if the client is not authenticated (see [RFC2791, section 3.1](https://tools.ietf.org/html/rfc2971#section-3.1)). Ideally, `client` would provide a `ErrNotConnected` error for commands that don't even require the user to be logged in.

Reply:
Oh, my bad. You're right, apologies.